lib/url_to_subscriber_list_criteria.rb

```diff
@@ -0,0 +1,149 @@
+require 'uri'
+
+class UrlToSubscriberListCriteria
+  MISSING_LOOKUP = "*** MISSING KEY ***"
+  EMAIL_SUPERTYPE = "email_document_supertype"
+  GOVERNMENT_SUPERTYPE = "government_document_supertype"
+  class UnprocessableUrl < StandardError; end
+
+  attr_reader :missing_lookup
+
+  def initialize(url, static_data = StaticData)
+    @url = URI.parse(url.strip)
+    @static_data = static_data
+    @missing_lookup = false
+  end
+
+  def convert
+    hash = map_url_to_hash.dup
+    if hash["links"]
+      links = hash["links"].each_with_object({}) do |(key, values), result|
+        result[key] = values.map { |value| lookup_content_id(key, value) }
+      end
+      hash.merge!("links" => links)
+    end
+    hash
+  end
+
+  def map_url_to_hash
+    @map_url_to_hash ||= begin
+      result = if @url.path == "/government/feed" || @url.path == "/government/feed.atom"
+                 { 'links' => from_params }
+               elsif (result = @url.path.match(%r{^/government/world/(.*)\.atom$}))
+                 { "links" => from_params.merge("world_locations" => [result[1]]) }
+               elsif (result = @url.path.match(%r{^/government/topical-events/(.*)\.atom$}))
+                 { "links" => from_params.merge("topical_events" => [result[1]]) }
+               elsif (result = @url.path.match(%r{^/government/people/(.*)\.atom$}))
+                 { "links" => from_params.merge("people" => [result[1]]) } # this is not yet implemented
+               elsif (result = @url.path.match(%r{^/government/organisations/(.*)\.atom$}))
+                 { "links" => from_params.merge("organisations" => [result[1]]) }
+               elsif (result = @url.path.match(%r{^/government/topics/(.*)\.atom$}))
+                 { "links" => from_params.merge(topic_map([result[1]]) => [result[1]]) }
```

Comment:
As discussed, don't believe this is required but will leave in as not 100% sure. I will group the ones which are migration related (old url format) at the bottom with a message to delete post-migration